specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2024-06-30/cloudhsm.json

```diff
@@ -0,0 +1,1430 @@
+        "Creating",
+        "Deleting",
+        "Failed",
+        "Updating",
+        "InternalError",
+        "Canceled"
+      ],
+      "x-ms-enum": {
+        "name": "PrivateEndpointConnectionProvisioningState",
+        "modelAsString": true
+      }
+    },
+    "PrivateEndpointConnectionListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "Array of private endpoint connections.",
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnection"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of private endpoint connections."
+        }
+      },
+      "description": "List of private endpoint connections associated with the specified resource."
+    },
+    "BackupRequestProperties": {
+      "type": "object",
+      "description": "Backup properties",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BackupRestoreRequestBaseProperties"
+        }
+      ]
+    },
+    "RestoreRequestProperties": {
+      "type": "object",
+      "description": "Cloud Hsm Cluster restore information",
+      "properties": {
+        "backupId": {
+          "type": "string",
+          "description": "The ID of the backup."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/BackupRestoreRequestBaseProperties"
+        }
+      ]
+    },
+    "BackupRestoreRequestBaseProperties": {
+      "type": "object",
+      "description": "Backup and Restore Common properties",
+      "properties": {
+        "azureStorageResourceUri": {
+          "type": "string",
+          "description": "The Azure blob storage container Uri which contains the backup"
+        },
+        "token": {
+          "type": "string",
+          "description": "The SAS token pointing to an Azure blob storage container. This property is reserved for Azure Backup Service."
+        }
+      },
+      "required": [
+        "azureStorageResourceUri"
+      ]
+    },
+    "BackupResponse": {
+      "type": "object",
+      "properties": {
+        "azureStorageResourceUri": {
+          "type": "string",
+          "description": "The Azure blob storage container Uri which contains the backup"
+        },
+        "backupId": {
+          "type": "string",
+          "description": "The ID of the backup."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/BackupRestoreBaseResponse"
+        }
+      ],
+      "description": "Backup operation Result"
+    },
+    "RestoreResponse": {
+      "type": "object",
+      "description": "Restore operation properties",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BackupRestoreBaseResponse"
+        }
+      ]
+    },
+    "BackupRestoreBaseResponse": {
+      "type": "object",
+      "properties": {
+        "status": {
+          "type": "string",
```

Comment:
As per feedback from Azure Backup, we should make it enum

Reply:
Thanks ! updated the property with Enum Values: InProgress, Succeeded, Failed